web/src/app/components/content/calculator/CalculateFormContainer.tsx

```diff
@@ -59,24 +61,36 @@ export const CalculateFormContainer: React.FunctionComponent<any> = () => {
   });
 
   return (
-    <form onSubmit={(e) => {
-      e.preventDefault();
-      addCalculation().then((r: any) => {
-        history.push('/calculation/' + r.data.addCalculation.id);
-      });
-    }}>
-      <ShipSection ship={ship} setShip={setShip} />
-      <LiquidSection liquid={liquid} setLiquid={setLiquid} />
-      <TransportSection transport={transport} setTransport={setTransport}></TransportSection>
-      <StandardSection standard={standard} setStandard={setStandard}></StandardSection>
-      <StandardButton
-        icon={EIcon.SUBMIT}
-        text={"Compute"}
-        style={{ margin: "30px 0 0 0" }}
-      ></StandardButton>
-      {/* <CalculationContainer
-        style={{ margin: "30px 0 0 0" }}
-      ></CalculationContainer> */}
-    </form>
+    <>
+      {!loading &&
+        <form onSubmit={(e) => {
+          e.preventDefault();
+          addCalculation().then((r: any) => {
+            history.push('/calculation/' + r.data.addCalculation.id);
```

Comment:
I think we can use onCompleted option for this, instead of resolving the promise.
https://www.apollographql.com/docs/react/data/queries/#options

Reply:
ok